src/openrct2/ride/Track.h

```diff
@@ -239,270 +239,270 @@ enum
     TRACK_ELEM_FLAG_BANKED = (1 << 15), // Also set on Spinning Tunnel and Log Flume reverser, probably to save a flag.
 };
 
-enum
+namespace TrackElemType
```

Comment:
Is there a reason for not having this inside an enum?

Reply:
These enums are used a lot, and also in some weird ways, which would mean a _lot_ of static_casts and a lot more refactoring:

* Used as an array index for 9 very commonly used arrays in `TrackData.h`
* Used as part of a subset of the `TRACK_CURVE_*` enums meaning it gets binary ORed with integers a lot

I'm not against converting it to an enum class, but since it's used so widely, I thought it might be better to do it incrementally instead of one large PR that changes half the codebase. This PR gets the work of renaming it done without changing any of the logic/types.

Happy to set it to draft continue working on it if you'd prefer, but it already has more than 4x more line changes than the JSON library PR 😅

Reply:
I guess its fine for now, one of the PRs introduced the EnumToFlags function, also with the EnumValue function a lot of static casts would be also avoided, so yeah lets do that in another PR.